### DISCUSSION TITLE
MCXN947 USB Device High speed issue

### DISCUSSION THREAD

**Discussion by mcuenthusiast** [hathach/tinyusb#2153] [Q&A]
Hi All,

I am using tinyusb to implement cdc class for high speed port in MCX(MCXN947) series controller.
For that  I have used below source files, 

src/portable/chipidea/ci_hs/hcd_ci_hs.c
src/portable/chipidea/ci_hs/dcd_ci_hs.c
src/portable/chipidea/ci_hs/hcd_ci_mcx.h
src/portable/chipidea/ci_hs/hcd_ci_type.h

I have used the board support package from below path

https://github.com/hathach/tinyusb/tree/master/hw/bsp/mcx

I am attaching the config file which we have used for this implementation.

Could some one please let me know which are the macros I need to define in config.h file.

I am able to implement the usb fs CDC by using hcd_ci_fs.c and dcd_ci_fs.c.

I have also attached the log file and windows error screenshots.
[tusb_config.txt](https://github.com/hathach/tinyusb/files/12102877/tusb_config.txt)
![windows_device_mgr_1](https://github.com/hathach/tinyusb/assets/132841642/b5d0d95e-41b4-439e-a367-dc4fb74b2339)
![windows_device_mgr_2](https://github.com/hathach/tinyusb/assets/132841642/6532bf97-7e05-405e-b1af-e3a8fcea7b20)
[tinyUSB_log.txt](https://github.com/hathach/tinyusb/files/12102886/tinyUSB_log.txt)
![windows_device_mgr_3](https://github.com/hathach/tinyusb/assets/132841642/a8660f50-3d96-4a28-a9f7-f3b1d271d92b)

**Comment by hathach** [comment]
did that stock example with makefile work for you ? if yes, you can get build info from there. It is alwayys easier to start with something that works first.

**Reply by mcuenthusiast** [reply]
Thanks for the prompt response @hathach, It's little difficult to incorporate Make file from stock example with my current development setup.

I was able to find the above issue and it's working fine now.

I defined below macro in config.h file and able to get the usb device enumerated on PC.
#define BOARD_TUD_MAX_SPEED OPT_MODE_HIGH_SPEED

But not I am not print anything on console, while further debugging I found that. 

DTR and RTS are not getting set, the if condition in attached image is not getting true.

![cdc_issue](https://github.com/hathach/tinyusb/assets/132841642/04a0d94b-0a7a-4aec-959b-2fa0225215ec)

**Reply by hathach** [reply]
Your teminal doesnt set dtr rts like others. Either change terminal or update firmwwre to skip connected() check

PS: you dont need to integrate any makefile, just make sure you test with a known good example and comparing with it first.

**Comment by mcuenthusiast** [comment]
Hi,

Any idea when this condition may occur?
![hs_cdc_issue](https://github.com/hathach/tinyusb/assets/132841642/f1972d0c-1f89-4f2f-bc72-aae21e04539c)

**Reply by hathach** [reply]
what is it since I am not familiar with the software ? does it happen with stock example ?
